typescript/sdk/src/core/HyperlaneRelayer.ts

```diff
@@ -206,44 +276,92 @@ export class HyperlaneRelayer {
     );
   }
 
-  start(chains = this.core.chains()): void {
-    assert(!this.stopRelayingHandler, 'Relayer already started');
+  protected async flushBacklog(): Promise<void> {
+    while (this.stopRelayingHandler) {
```

Comment:
This infinite loop approach is okay when the only consumer of this is the CLI but if were to ever use this in UI, we would want a design that relays when consumers call in. Otherwise, this could block the UI thread. In general, this loop-and-watch behavior _feels_ like it belongs in the CLI IMO, whereas an SDK class should be more flexible as to how it's used

Reply:
> we would want a design that relays when consumers call in

unless I misunderstand what you mean by this, the existing `relayMessage` function does what you are describing

>  In general, this loop-and-watch behavior feels like it belongs in the CLI IMO

interesting, im not really sure how that would look
it seems like it would require a lot of "internal" knowledge for clients to implement the behavior you are describing

I quite like the current interface of
```js
relayer.start()
process.once('SIGINT', () => relayer.stop())
```